addons/website_sale/controllers/main.py

```diff
@@ -1541,15 +1526,16 @@ def shop_payment(self, **post):
         if redirection := self._check_cart_and_addresses(order_sudo):
             return redirection
 
+        order_sudo._recompute_taxes()
+        order_sudo._recompute_prices()
         render_values = self._get_shop_payment_values(order_sudo, **post)
         render_values['only_services'] = order_sudo and order_sudo.only_services
 
         if render_values['errors']:
             render_values.pop('payment_methods_sudo', '')
             render_values.pop('tokens_sudo', '')
 
-        # As the initial page sending us to payment is /shop/confirm_order
-        render_values.update(request.website._get_checkout_step_values('/shop/confirm_order'))
+        render_values.update(request.website._get_checkout_step_values())
```

Comment:
We can take this opportunity to remove `href` from the args of `_get_checkout_step_values` now that it's not used anywhere else anymore.
https://github.com/odoo/odoo/blob/master/addons/website_sale/models/website.py#L796-L797
```diff
-    def _get_checkout_step_values(self, href=None):
-        href = href or request.httprequest.path
+    def _get_checkout_step_values(self):
+        href = request.httprequest.path
```